src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComponentResourceManager.cs

```diff
@@ -149,10 +149,10 @@ public virtual void ApplyResources(object value, string objectName, CultureInfo
                     }
                 }
 
-                // Character after objectName.Length should be a ".", or else we should continue.
+                // Character after objectName.Length should be a "." or a '-', or else we should continue.
```

Comment:
This potentially breaks anyone already using ComponentResourceManager over resources containing hyphens right?